src/Plugin/Field/FieldFormatter/MediaImageDisplayFormatter.php

```diff
@@ -139,18 +140,20 @@ public function settingsForm(array $form, FormStateInterface $form_state) {
       '#options' => [
         'nothing' => $this->t('Nothing'),
 //        'content' => $this->t('Content'),
-        'media' => $this->t('Media')],
+        ($this->getFieldList($mediaFieldsList, 'link')) ?? 'media' => $this->t('Media')],
     ];
 
     $element['media_link_field'] = [
       '#title' => t('Media Link field'),
       '#type' => 'select',
-      '#required' => TRUE,
       '#default_value' => $this->getSetting('media_link_field'),
       '#options' => $this->getFieldList($mediaFieldsList, 'link'),
       '#states' => [
         'visible' => [
-          ':input[name="fields[field_visual][settings_edit_form][settings][link_source]"]' => array('value' => 'media')
+          ':input[name="fields['.$this->fieldDefinition->getName().'][settings_edit_form][settings][link_source]"]' => array('value' => 'media')
```

Comment:
$this->fieldDefinition->getName() should be in a variable because it's called multiple times